lib/pyfrc/test_support/pytest_plugin.py

```diff
@@ -59,6 +59,9 @@ def pytest_runtest_setup(self):
                        "If your robot class has an __init__ function, it must call super().__init__()!"
     
     def pytest_runtest_teardown(self, nextitem):
+        # Let any child threads run in realtime to allow cancelling if it
+        # has been implemented.
+        self._fake_time.freerun_children()
```

Comment:
Yeah, thinking about it on the way home I agree with you.
Let's make the tests fail if code spawns a thread but doesn't clean it up nicely at the end - enforce good housekeeping where we can.
I'll try to write some unit tests over this stuff to demonstrate the behaviours. Not sure how easy it will be, but I'll give it a go.